components/clp-package-utils/clp_package_utils/scripts/native/archive_manager.py

```diff
@@ -0,0 +1,365 @@
+    )
+
+    # Delete by filter arguments
+    del_filter_parser.add_argument(
+        "begin_ts",
+        type=int,
+        help="Time-range lower-bound (inclusive) as milliseconds from the UNIX epoch.",
+    )
+    del_filter_parser.add_argument(
+        "end_ts",
+        type=int,
+        help="Time-range upper-bound (include) as milliseconds from the UNIX epoch.",
+    )
+
+    parsed_args = args_parser.parse_args(argv[1:])
+
+    # Validate and load config file
+    config_file_path = Path(parsed_args.config)
+    try:
+        clp_config = load_config_file(config_file_path, default_config_file_path, clp_home)
+        clp_config.validate_logs_dir()
+    except:
+        logger.exception("Failed to load config.")
+        return -1
+
+    database_config = clp_config.database
+    archives_dir = clp_config.archive_output.get_directory()
+    if not archives_dir.exists():
+        logger.error("`archive_output.directory` doesn't exist.")
+        return -1
+
+    if FIND_COMMAND == parsed_args.subcommand:
+        return _find_archives(
+            archives_dir,
+            database_config,
+            parsed_args.begin_ts,
+            parsed_args.end_ts,
+        )
+    elif DEL_COMMAND == parsed_args.subcommand:
+        if BY_IDS_COMMAND == parsed_args.del_subcommand:
+            return _delete_archives_by_ids(
+                archives_dir,
+                database_config,
+                parsed_args.ids,
+                parsed_args.dry_run,
+            )
+        elif BY_FILTER_COMMAND == parsed_args.del_subcommand:
+            return _delete_archives_by_filter(
+                archives_dir,
+                database_config,
+                parsed_args.begin_ts,
+                parsed_args.end_ts,
+                parsed_args.dry_run,
+            )
+
+
+def _find_archives(
+    archives_dir: Path,
+    database_config: Database,
+    begin_ts: int,
+    end_ts: int = None,
+) -> int:
+    """
+    Lists all archive IDs, if begin_its and end_its are provided,
+    only lists archives where `begin_ts <= archive.begin_timestamp` and
+    `archive.end_timestamp <= end_ts`.
+    :param archives_dir:
+    :param database_config:
+    :param begin_ts:
+    :param end_ts:
+    """
+    archive_ids: List[str]
+    logger.info("Starting to find archives from the database.")
+    try:
+        sql_adapter = SQL_Adapter(database_config)
+        clp_db_connection_params = database_config.get_clp_connection_params_and_type(True)
+        table_prefix = clp_db_connection_params["table_prefix"]
+        with closing(sql_adapter.create_connection(True)) as db_conn, closing(
+            db_conn.cursor(dictionary=True)
+        ) as db_cursor:
+
+            params = (begin_ts,)
+            query = f"SELECT id FROM `{table_prefix}archives` WHERE begin_timestamp >= %s"
+            if end_ts is not None:
+                query += " AND end_timestamp <= %s"
+                params = (begin_ts, end_ts)
```

Comment:
any chance we can "append" to the tuple?
```suggestion
                params = params  + (end_s, )
```